config/knex/index.js

```diff
@@ -1,20 +1,18 @@
 let knex = require('knex');
 
-const environment = process.env.NODE_ENV || 'development';
+const environment = process.env.NODE_ENV || 'local';
 const configuration = require('../../knexfile')[environment];
 knex = require('knex')(configuration);
 
 function connectToDB() {
   let retries = 5;
-  let connected = false;
-  if (!connected && retries > 0) {
+  if (retries > 0) {
     knex.raw('select 1+1 as result').catch((err) => {
       console.error("Couldn't connect to DB", err);
       retries -= 1;
       console.error(`Retries left: ${retries}`);
       setTimeout(connectToDB, 5000);
     });
-    connected = true;
   }
```

Comment:
Why the need for this change? The idea was to prevent it from retrying to connect up to 5x if it was already connected. Seems unnecessary to remove.

Reply:
Hi @samc621, thanks, this is as you say my first ever open source PR! 
Yes I understand why you had the connected variable there, but I removed it as its scope is local, so when you first go into this block, 'connected' would always be set to false (so you would never know if a connection was actually active), additionally the if block `knex.raw` catch handles the retry if the connection is unsuccessful and calls the whole function recursively.
I am a backend developer and exploring the front-end so would be happy to be educated if I am wrong.
I did not see anywhere else in the codebase that uses or calls this either.

Reply:
Thinking about it even further, as the retries is block scope, the code should have the following updates:
Add retries as a parameter and default to 5 - `function connectToDB(retries = 5) {`
Remove the now redundant retires - `let retries = 5;`
Provide the new retries value when calling the function after timeout - `setTimeout(connectToDB, 5000, retries);`

Reply:
Because of the require stack, this function gets called on server startup. I think my comment still stands true, this change will cause it to retry the connection (by running this query) 5x, right? The reason for the `connected` variable is to check if it is connected so that it doesn't keep retrying if there was no error caught.

Reply:
No this should not be the case, the function gets called once on startup, there is no loop in the code so the only time the function would get called again is if a connection to the db is attempted `knex.raw('select 1+1 as result')` and the result fails.
If the result fails then it will go into the catch block and call the same function again (albeit with retries minus 1):
```
function connectToDB(retries = 5) {  
  if (retries > 0) {
    knex.raw('select 1+1 as result').catch((err) => {
      ...
      setTimeout(connectToDB, 5000, retries);
    }
}
```

The `connected` variable did nothing before as it was only set to true just before the function exit, after that it was gone from memory so was in fact redundant.

Reply:
Oh yes I see now, I misread the code. Thanks!